lmnet/tests/lmnet_tests/metrics_tests/test_object_keypoint_similarity.py

```diff
@@ -0,0 +1,180 @@
+    joints_pred[0, 1] = 70
+    joints_pred[0, 2] = 1
+
+    joints_pred[2, 0] = 1000
+    joints_pred[2, 1] = 1000
+    joints_pred[2, 2] = 1
+
+    expected = 0.2358359
+
+    result = compute_oks(joints_gt, joints_pred, image_size)
+
+    assert np.allclose(result, expected)
+
+    # case2
+
+    joints_gt = np.zeros((17, 3))
+    joints_pred = np.zeros((17, 3))
+    image_size = (160, 160)
+
+    joints_gt[0, 0] = 80
+    joints_gt[0, 1] = 80
+    joints_gt[0, 2] = 0
+
+    joints_pred[0, 0] = 70
+    joints_pred[0, 1] = 70
+    joints_pred[0, 2] = 1
+
+    joints_pred[2, 0] = 1000
+    joints_pred[2, 1] = 1000
+    joints_pred[2, 2] = 1
+
+    expected = -1
+
+    result = compute_oks(joints_gt, joints_pred, image_size)
+
+    assert np.allclose(result, expected)
+
+    # case3
+
+    joints_gt = np.zeros((17, 3))
+    joints_pred1 = np.zeros((17, 3))
+    joints_pred2 = np.zeros((17, 3))
+    image_size = (160, 160)
+
+    joints_gt[0, 0] = 80
+    joints_gt[0, 1] = 80
+    joints_gt[0, 2] = 1
+
+    joints_pred1[0, 0] = 70
+    joints_pred1[0, 1] = 70
+    joints_pred1[0, 2] = 1
+
+    joints_pred2[0, 0] = 78
+    joints_pred2[0, 1] = 78
+    joints_pred2[0, 2] = 1
+
+    result1 = compute_oks(joints_gt, joints_pred1, image_size)
+    result2 = compute_oks(joints_gt, joints_pred2, image_size)
+
+    assert result2 > result1
+
+
+def test_compute_oks_batch():
+
+    # case1
+
+    joints_gt = np.zeros((1, 17, 3))
+    joints_pred = np.zeros((1, 17, 3))
+    image_size = (160, 160)
+
+    joints_gt[0, 0, 0] = 80
+    joints_gt[0, 0, 1] = 80
+    joints_gt[0, 0, 2] = 1
+
+    joints_pred[0, 0, 0] = 70
+    joints_pred[0, 0, 1] = 70
+    joints_pred[0, 0, 2] = 1
+
+    expected = 0.2358359
+
+    result = compute_oks_batch(joints_gt, joints_pred, image_size)
+
+    assert np.allclose(result, expected)
+
+    # case2
+
+    joints_gt = np.zeros((2, 17, 3))
+    joints_pred = np.zeros((2, 17, 3))
+    image_size = (160, 160)
+
+    joints_gt[0, 0, 0] = 80
+    joints_gt[0, 0, 1] = 80
+    joints_gt[0, 0, 2] = 1
+
+    joints_pred[0, 0, 0] = 70
+    joints_pred[0, 0, 1] = 70
+    joints_pred[0, 0, 2] = 1
+
+    joints_gt[1, 0, 0] = 50
+    joints_gt[1, 0, 1] = 50
+    joints_gt[1, 0, 2] = 1
+
+    joints_pred[1, 0, 0] = 50
+    joints_pred[1, 0, 1] = 50
+    joints_pred[1, 0, 2] = 1
+
+    expected = 0.61791795
+
+    result = compute_oks_batch(joints_gt, joints_pred, image_size)
+
+    assert np.allclose(result, expected)
+
+    # case3
+
+    joints_gt = np.zeros((2, 17, 3))
+    joints_pred = np.zeros((2, 17, 3))
+    image_size = (160, 160)
+
+    joints_gt[0, 0, 0] = 80
+    joints_gt[0, 0, 1] = 80
+
+    joints_pred[0, 0, 0] = 70
+    joints_pred[0, 0, 1] = 70
+    joints_pred[0, 0, 2] = 1
+
+    joints_gt[1, 0, 0] = 50
+    joints_gt[1, 0, 1] = 50
+
+    joints_pred[1, 0, 0] = 50
+    joints_pred[1, 0, 1] = 50
+    joints_pred[1, 0, 2] = 1
+
+    try:
+        compute_oks_batch(joints_gt, joints_pred, image_size)
```

Comment:
This test case is for exception?
Is it possible to use `pytest.raises` ?
http://doc.pytest.org/en/latest/assert.html

Reply:
I think both of them are okay. I prefer native python ways.